packages/shared/lib/contexts/onboarding/stores/shimmer-claiming-transactions.store.ts

```diff
@@ -0,0 +1,44 @@
+import { get, Writable } from 'svelte/store'
+
+import { InvalidTransactionIdError } from '@core/wallet'
+import { persistent } from '@lib/helpers'
+
+import { IShimmerClaimingTransactionStore } from '../interfaces'
+
+import { onboardingProfile } from './onboarding-profile.store'
+
+export const shimmerClaimingTransactions: Writable<IShimmerClaimingTransactionStore> = persistent(
+    'shimmerClaimingTransactions',
+    {}
+)
+
+export function isShimmerClaimingTransaction(transactionId: string, profileId?: string): boolean {
+    const _shimmerClaimingTransactions = get(shimmerClaimingTransactions)
+    profileId = profileId ? profileId : get(onboardingProfile)?.id
+    return profileId in _shimmerClaimingTransactions && _shimmerClaimingTransactions[profileId][transactionId]
+}
+
+export function persistShimmerClaimingTransaction(transactionId: string, profileId?: string): void {
+    if (!transactionId) {
+        throw new InvalidTransactionIdError()
+    }
+
+    profileId = profileId ? profileId : get(onboardingProfile)?.id
+    shimmerClaimingTransactions.update((_shimmerClaimingTransactions) => {
+        if (profileId in _shimmerClaimingTransactions) {
+            _shimmerClaimingTransactions[profileId][transactionId] = true
+        } else {
+            _shimmerClaimingTransactions[profileId] = { [transactionId]: true }
+        }
```

Comment:
i think we can drop the first if statement completely
```suggestion
        _shimmerClaimingTransactions[profileId] = { [transactionId]: true }
```

Reply:
Just this statement would end up overwriting any other claiming transactions that were also added for other accounts (within the same profile), IIUC

Reply:
youre right